diff --git a/arch/arm64/boot/dts/allwinner/sun50i-h618-orangepi-zero3.dts b/arch/arm64/boot/dts/allwinner/sun50i-h618-orangepi-zero3.dts
index 4ff1327ec..dd55ad49c 100644
--- a/arch/arm64/boot/dts/allwinner/sun50i-h618-orangepi-zero3.dts
+++ b/arch/arm64/boot/dts/allwinner/sun50i-h618-orangepi-zero3.dts
@@ -119,7 +119,7 @@

 &gpu {
 	mali-supply = <&reg_dcdc1>;
-	status = "disabled";
+	status = "okay";
 };

 &emac0 {
